### ISSUE TITLE
Improve search results code viewing accessibility

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#19272]
## Search results

Example page: https://sourcegraph.com/search?q=useState&patternType=regexp

Screenshot:
![image](https://user-images.githubusercontent.com/9516420/111636322-45ed1c00-87f0-11eb-8f68-05b04055c784.png)

### Known issues
1. Focus states. Try to navigate down the results with just a keyboard, you will realise that the focus seems to be 'lost' once it reaches some code. This seems to be because we do not show any focus online on those areas.

2. The "title bar"
![image](https://user-images.githubusercontent.com/9516420/111636579-88165d80-87f0-11eb-891c-dd133cb25f2f.png)
This bar is essentially a `<div>` with an onClick for toggle functionality. The "Show X more matches" text should be the actual button here. This will ensure that it is possible to actually navigate to this button with a keyboard and properly support screenreaders.

3. On the diff view, it is possible to anchor directly to specific lines. It's not possible to navigate to these as a keyboard.
![image](https://user-images.githubusercontent.com/9516420/112169734-5f280b00-8bea-11eb-9b58-f46329e551e7.png)

4. On the diff view, there is currently no way for a screen reader to understand if a line has been added/deleted/maintained.

**Comment by umpox** [comment]
I think this better fits/is more relevant to the search team redesign, please let me know if I'm wrong!
